packages/salesforce-adapter/src/filters/utils.ts

```diff
@@ -252,3 +254,37 @@ export const isInstanceOfTypeChange = (type: string) => (
     isInstanceOfType(type)(getChangeElement(change))
   )
 )
+
+export const toSaltoWarning = (message: string): SaltoError => (
+  ({
+    message,
+    severity: 'Warning',
+  })
+)
+
+export const ensureSafeFilterFetch = (
+  fetchFilterFunc: Required<Filter>['onFetch'],
+  warningMessage: string,
+): Required<Filter>['onFetch'] =>
+  async elements => {
+    try {
+      return await fetchFilterFunc(elements)
+    } catch (e) {
+      log.error('Failed to run filter with the following error %o, stack %o', e, e.stack)
+      return {
+        errors: [toSaltoWarning(warningMessage)],
+      }
+    }
+  }
+export const ensureFilterEnabled = (
+  fetchFilterFunc: Required<Filter>['onFetch'],
+  filterName: keyof OptionalFeatures,
+  config : FilterContext,
+): Required<Filter>['onFetch'] =>
+  async elements => {
+    if (!config.fetchProfile.isFeatureEnabled(filterName)) {
+      log.debug('skipping %o filter due to configuration', filterName)
+      return undefined
+    }
+    return fetchFilterFunc(elements)
+  }
```

Comment:
I'm still probably missing context, but it looks like these two are only used together so far, and the `toSaltoWarning` parameters are identical except for something describing the filter (which I assume can also just be the filter name?)
wdyt about merging them into one? can be something like 
```suggestion
export const ensureSafeFilterFetch = (
  fetchFilterFunc: Required<Filter>['onFetch'],
  filterName: keyof OptionalFeatures,
  config : FilterContext,
): Required<Filter>['onFetch'] =>
  async elements => {
    try {
      if (!config.fetchProfile.isFeatureEnabled(filterName)) {
        log.debug('skipping %o filter due to configuration', filterName)
        return undefined
      }
      return fetchFilterFunc(elements)
    } catch (e) {
      log.error('Failed to run filter with the following error %o, stack %o', e, e.stack)
      return {
        errors: [({
          message: `Failed to run filter ${filterName} due to unexpected error`,
          severity: 'Warning',
        })],
      }
    }
  }
```

Reply:
Cool! We thought to leave them separate. Currently they are only used together but they don't have to be.